src/neural/cuda/layers.cc

```diff
@@ -1225,6 +1141,424 @@ ResidualBlock<DataType>::~ResidualBlock() {
   }
 }
 
+template <typename DataType>
+void allocAndUpload(DataType** gpu_dest, std::vector<float> cpu_src,
+                    void* scratch) {
+  size_t size = cpu_src.size() * sizeof(DataType);
+  if (size == 0) {
+    *gpu_dest = nullptr;
+    return;
+  }
+  ReportCUDAErrors(cudaMalloc(gpu_dest, size));
+  ReportCUDAErrors(
+      cudaMemcpy(scratch, &cpu_src[0], size, cudaMemcpyHostToDevice));
+  copyTypeConverted((DataType*)(*gpu_dest), (float*)scratch,
+                    (int)cpu_src.size(), 0);
+}
+
+template <typename DataType>
+AttentionPolicyHead<DataType>::AttentionPolicyHead(BaseLayer<DataType>* ip,
+                                                   const LegacyWeights& weights,
+                                                   void* scratch)
+    : BaseLayer<DataType>(ip->GetC(), 8, 8, ip) {
+  embedding_op_size_ = weights.ip_pol_b.size();
+  wq_op_size_ = weights.ip2_pol_b.size();
+  wk_op_size_ = weights.ip3_pol_b.size();
+  ppo_op_size_ = weights.ip4_pol_b.size();
```

Comment:
This is unused - but also ppo has no bias so the unused value calculated here is technically wrong.

Reply:
updated to new net.proto, and removed them.